### PR TITLE
refactor the way we fetch and unblind utxos

### LINKED ISSUE #2: Improve fetchAndUnblindTxs
- [ ] Let the user to give us a mapping of addresses/scripts and the relative blinding keys
- [ ] Refactor to expose a streams API to let the frontend developers to consume as soon a single transaction is unblinded

### PR DESCRIPTION
**Pull request by louisinger**
**This PR proposes a refacto for `FetchAndUnblindUtxos` function**

TL;DR
- `FetchAndUnblindUtxos` now accepts an array of `AddressWithBlindingKey` as arguments
- `FetchAndUnblindUtxosGenerator` implements a javascript generator to optimize the way we fetch  and unblind the utxos
- _Unblinding_ step is now a nonblocking step. The idea is to return the original utxos if the unblind step fails (thus, the function handles unconfidential utxos)
- `BlindingKeyGetter` is a new type defining a function `string => string | undefined`
- `FetchAndUnblindTxs` returns all the txs for a list of addresses, and unblind them using a `BlindingKeyGetter`
- `FetchAndUnblindTxsGenerator` implements a js generator to optimize the way we fetch and unblind the transactions.

**I want all the utxos for a given address:**
```js
const utxos = await fetchAndUnblindUtxos(
    [{ address, blindingKey: blindPrivKey }],
    url
  );
```

**I want the first two utxos:**
```js
const utxosGenerator = fetchAndUnblindUtxosGenerator(
[{  
   address: senderAddress,
   blindingKey: sender.getNextAddress().blindingPrivateKey,
}],  
   APIURL
);

const utxo0 = (await utxosGenerator.next()).value
const utxo1 = (await utxosGenerator.next()).value
```

Remarks: 
- the generator returns the total number of utxos fetched while it is done.
- the explorer is called only if the generator's iterator reaches the next address's utxos.

@tiero  @altafan please review

it closes #2

### CODE REVIEW

**Comment by louisinger** [comment]
description updated + conflicts merged @tiero
